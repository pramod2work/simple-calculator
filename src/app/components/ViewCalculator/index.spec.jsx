/* global describe test expect beforeEach */

import React from 'react'
import { shallow } from 'enzyme'

import CalculatorKeyPad from './index'

describe('<CalculatorKeyPad />', () => {
  describe('Shallow Rendering', () => {
    let component
    beforeEach(() => {
      component = shallow(<CalculatorKeyPad />)
    })

    test('snapshot test', () => {
      expect(component).toMatchSnapshot()
    })

    test('Check for values selected', () => {
      component.find('[data-test-id="key-1"]').last().simulate('click')
      expect(component.state().numSelected).toBe('1')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      expect(component.state().numSelected).toBe('')
      component.find('[data-test-id="key-2"]').last().simulate('click')
      expect(component.state().numSelected).toBe('2')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-3"]').last().simulate('click')
      expect(component.state().numSelected).toBe('3')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-4"]').last().simulate('click')
      expect(component.state().numSelected).toBe('4')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-5"]').last().simulate('click')
      expect(component.state().numSelected).toBe('5')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-6"]').last().simulate('click')
      expect(component.state().numSelected).toBe('6')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-7"]').last().simulate('click')
      expect(component.state().numSelected).toBe('7')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-8"]').last().simulate('click')
      expect(component.state().numSelected).toBe('8')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-9"]').last().simulate('click')
      expect(component.state().numSelected).toBe('9')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-0"]').last().simulate('click')
      expect(component.state().numSelected).toBe('0')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-+"]').last().simulate('click')
      expect(component.state().numSelected).toBe('+')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key--"]').last().simulate('click')
      expect(component.state().numSelected).toBe('-')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-*"]').last().simulate('click')
      expect(component.state().numSelected).toBe('*')
      component.find('[data-test-id="key-D"]').last().simulate('click')
      component.find('[data-test-id="key-/"]').last().simulate('click')
      expect(component.state().numSelected).toBe('/')
    })

    test('Check for Addition', () => {
      component.find('[data-test-id="key-1"]').last().simulate('click')
      component.find('[data-test-id="key-+"]').last().simulate('click')
      component.find('[data-test-id="key-2"]').last().simulate('click')
      component.find('[data-test-id="key-="]').last().simulate('click')
      expect(component.state().numSelected).toBe(3)
    })

    test('handle keyboard event', () => {
      component.instance().handleKeyboardEvent({ key: 1 })
      component.instance().handleKeyboardEvent({ key: 3 })
      expect(component.state().numSelected).toBe('13')
      component.instance().handleKeyboardEvent({ key: 'Del' })
      expect(component.state().numSelected).toBe('')
    })
  })
})
