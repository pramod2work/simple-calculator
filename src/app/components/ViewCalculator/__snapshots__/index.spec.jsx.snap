// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorKeyPad /> Shallow Rendering snapshot test 1`] = `
<div>
  <styled.div
    data-test-id="display-section"
  >
    <input
      aria-label="calculator operation output, select value and operation from below keypad section"
      disabled={true}
      id="calculator-field"
      name="calculatorField"
      type="text"
      value=""
    />
  </styled.div>
  <styled.div
    data-test-id="keypad-row-0"
    isLast={false}
    key="seq-row-1"
  >
    <styled.div
      key="seq-delete"
    >
      <styled.button
        data-test-id="key-D"
        isDelete={true}
        isLast={false}
        onClick={[Function]}
      >
        <span>
          AC
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-blank-1"
    >
      <Styled(styled.div)
        data-test-id="key-blank"
        isLast={false}
      />
    </styled.div>
    <styled.div
      key="seq-blank-2"
    >
      <Styled(styled.div)
        data-test-id="key-blank"
        isLast={false}
      />
    </styled.div>
    <styled.div
      key="seq-blank-3"
    >
      <Styled(styled.div)
        data-test-id="key-blank"
        isLast={true}
      />
    </styled.div>
  </styled.div>
  <styled.div
    data-test-id="keypad-row-1"
    isLast={false}
    key="seq-row-2"
  >
    <styled.div
      key="seq-1"
    >
      <styled.button
        data-test-id="key-1"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          1
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-2"
    >
      <styled.button
        data-test-id="key-2"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          2
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-3"
    >
      <styled.button
        data-test-id="key-3"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          3
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-+"
    >
      <styled.button
        data-test-id="key-+"
        isLast={true}
        onClick={[Function]}
      >
        <span>
          +
        </span>
      </styled.button>
    </styled.div>
  </styled.div>
  <styled.div
    data-test-id="keypad-row-2"
    isLast={false}
    key="seq-row-3"
  >
    <styled.div
      key="seq-4"
    >
      <styled.button
        data-test-id="key-4"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          4
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-5"
    >
      <styled.button
        data-test-id="key-5"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          5
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-6"
    >
      <styled.button
        data-test-id="key-6"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          6
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq--"
    >
      <styled.button
        data-test-id="key--"
        isLast={true}
        onClick={[Function]}
      >
        <span>
          -
        </span>
      </styled.button>
    </styled.div>
  </styled.div>
  <styled.div
    data-test-id="keypad-row-3"
    isLast={false}
    key="seq-row-4"
  >
    <styled.div
      key="seq-7"
    >
      <styled.button
        data-test-id="key-7"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          7
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-8"
    >
      <styled.button
        data-test-id="key-8"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          8
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-9"
    >
      <styled.button
        data-test-id="key-9"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          9
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-*"
    >
      <styled.button
        data-test-id="key-*"
        isLast={true}
        onClick={[Function]}
      >
        <span>
          *
        </span>
      </styled.button>
    </styled.div>
  </styled.div>
  <styled.div
    data-test-id="keypad-row-4"
    isLast={true}
    key="seq-row-5"
  >
    <styled.div
      key="seq-."
    >
      <styled.button
        data-test-id="key-."
        isLast={false}
        onClick={[Function]}
      >
        <span>
          .
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-0"
    >
      <styled.button
        data-test-id="key-0"
        isLast={false}
        onClick={[Function]}
      >
        <span>
          0
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-="
    >
      <styled.button
        data-test-id="key-="
        isLast={false}
        onClick={[Function]}
      >
        <span>
          =
        </span>
      </styled.button>
    </styled.div>
    <styled.div
      key="seq-/"
    >
      <styled.button
        data-test-id="key-/"
        isLast={true}
        onClick={[Function]}
      >
        <span>
          /
        </span>
      </styled.button>
    </styled.div>
  </styled.div>
</div>
`;
